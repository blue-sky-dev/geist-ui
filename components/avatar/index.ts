import Avatar from './avatar'
import { AvatarProps } from './avatar'

export type Props = AvatarProps
export default Avatar
