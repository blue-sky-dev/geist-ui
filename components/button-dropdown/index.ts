import ButtonDropdown from './button-dropdown'
import ButtonDropdownItem from './item'

ButtonDropdown.Item = ButtonDropdownItem

export default ButtonDropdown
