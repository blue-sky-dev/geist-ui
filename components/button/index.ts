import Button from './button'
import { ButtonProps } from './button'

export type Props = ButtonProps
export default Button
