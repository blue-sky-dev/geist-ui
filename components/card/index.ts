import Card from './card'
import { CardProps } from './card'

export type Props = CardProps
export default Card
