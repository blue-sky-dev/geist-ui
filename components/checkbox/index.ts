import Checkbox from './checkbox'
import CheckboxGroup from './checkbox-group'
import { CheckboxProps, CheckboxEvent } from './checkbox'

export type Props = CheckboxProps
export type Event = CheckboxEvent

Checkbox.Group = CheckboxGroup

export default Checkbox
