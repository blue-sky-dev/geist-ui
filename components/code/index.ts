import Code from './code'
import { CodeProps } from './code'

export type Props = CodeProps
export default Code
