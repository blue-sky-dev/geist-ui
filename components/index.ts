/// <reference types="styled-jsx" />

export { default as useTheme } from './styles/use-theme'
export { default as ZEITUIProvider } from './providers/zeit-ui-provider'
export { default as useToasts } from './toast'
export { default as useInput } from './input/use-input'
export { default as useModal } from './modal/use-modal'
export { default as useTabs } from './tabs/use-tabs'
export { default as CSSBaseline } from './styles/css-baseline'
export { default as Avatar } from './avatar'
export { default as Text } from './text'
export { default as Note } from './note'
export { default as Link } from './link'
export { default as Col } from './col'
export { default as Row } from './row'
export { default as Container } from './row'
export { default as Button } from './button'
export { default as Card } from './card'
export { default as Code } from './code'
export { default as Display } from './display'
export { default as Description } from './description'
export { default as Image } from './image'
export { default as Spacer } from './spacer'
export { default as Tag } from './tag'
export { default as Dot } from './dot'
export { default as Keyboard } from './keyboard'
export { default as Checkbox } from './checkbox'
export { default as Fieldset } from './fieldset'
export { default as Modal } from './modal'
export { default as Spinner } from './spinner'
export { default as ButtonDropdown } from './button-dropdown'
export { default as Capacity } from './capacity'
export { default as Input } from './input'
export { default as Radio } from './radio'
export { default as Select } from './select'
export { default as Tabs } from './tabs'
export { default as Progress } from './progress'
export { default as Tree } from './file-tree'
export { default as Badge } from './badge'
export { default as AutoComplete } from './auto-complete'
export { default as Collapse } from './collapse'
