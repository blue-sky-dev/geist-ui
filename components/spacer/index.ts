import Spacer from './spacer'
import { SpacerProps } from './spacer'

export type Props = SpacerProps
export default Spacer
