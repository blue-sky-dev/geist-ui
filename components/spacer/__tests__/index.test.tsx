import React from 'react'
import { mount } from 'enzyme'
import { Spacer } from 'components'

describe('Spacer', () => {
  it('should render correctly', () => {
    const wrapper = mount(<Spacer />)
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should support x and y', () => {
    const wrapper = mount(
      <div>
        <Spacer x={5} />
        <Spacer x={15} />
        <Spacer y={15} />
        <Spacer y={2} />
      </div>,
    )
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should work with float', () => {
    const wrapper = mount(
      <div>
        <Spacer x={2.2} />
        <Spacer x={1.5} />
        <Spacer y={0.1} />
        <Spacer y={1.8} />
      </div>,
    )
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should work with inline mode', () => {
    const wrapper = mount(<Spacer inline />)
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })
})
