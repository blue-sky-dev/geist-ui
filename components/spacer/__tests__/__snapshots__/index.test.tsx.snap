// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer should render correctly 1`] = `
"<span class=\\"\\"><style>
        span {
          display: block;
          height: 1px;
          width: 1px;
          margin-left: calc(15.25pt + 1px * 0);
          margin-top: calc(15.25pt + 1px * 0);
        }
      </style></span>"
`;

exports[`Spacer should support x and y 1`] = `
"<div><span class=\\"\\"><style>
        span {
          display: block;
          height: 1px;
          width: 1px;
          margin-left: calc(76.25pt + 1px * 4);
          margin-top: calc(15.25pt + 1px * 0);
        }
      </style></span><span class=\\"\\"><style>
        span {
          display: block;
          height: 1px;
          width: 1px;
          margin-left: calc(228.75pt + 1px * 14);
          margin-top: calc(15.25pt + 1px * 0);
        }
      </style></span><span class=\\"\\"><style>
        span {
          display: block;
          height: 1px;
          width: 1px;
          margin-left: calc(15.25pt + 1px * 0);
          margin-top: calc(228.75pt + 1px * 14);
        }
      </style></span><span class=\\"\\"><style>
        span {
          display: block;
          height: 1px;
          width: 1px;
          margin-left: calc(15.25pt + 1px * 0);
          margin-top: calc(30.5pt + 1px * 1);
        }
      </style></span></div>"
`;

exports[`Spacer should work with float 1`] = `
"<div><span class=\\"\\"><style>
        span {
          display: block;
          height: 1px;
          width: 1px;
          margin-left: calc(33.550000000000004pt + 1px * 1.2000000000000002);
          margin-top: calc(15.25pt + 1px * 0);
        }
      </style></span><span class=\\"\\"><style>
        span {
          display: block;
          height: 1px;
          width: 1px;
          margin-left: calc(22.875pt + 1px * 0.5);
          margin-top: calc(15.25pt + 1px * 0);
        }
      </style></span><span class=\\"\\"><style>
        span {
          display: block;
          height: 1px;
          width: 1px;
          margin-left: calc(15.25pt + 1px * 0);
          margin-top: calc(1.5250000000000001pt + 1px * -0.9);
        }
      </style></span><span class=\\"\\"><style>
        span {
          display: block;
          height: 1px;
          width: 1px;
          margin-left: calc(15.25pt + 1px * 0);
          margin-top: calc(27.45pt + 1px * 0.8);
        }
      </style></span></div>"
`;

exports[`Spacer should work with inline mode 1`] = `
"<span class=\\"\\"><style>
        span {
          display: inline-block;
          height: 1px;
          width: 1px;
          margin-left: calc(15.25pt + 1px * 0);
          margin-top: calc(15.25pt + 1px * 0);
        }
      </style></span>"
`;
